stock_partner_delivery_window/models/res_partner.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from collections import defaultdict
+from datetime import time
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.tools.misc import format_time
+
+from odoo.addons.partner_tz.tools import tz_utils
+
+WORKDAYS = list(range(5))
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    delivery_time_preference = fields.Selection(
+        [
+            ("anytime", "Any time"),
+            ("time_windows", "Fixed time windows"),
+            ("workdays", "Weekdays (Monday to Friday)"),
+        ],
+        string="Delivery time schedule preference",
+        default="workdays",
```

Comment:
@grindtildeath @mmequignon the default here should be changed?

Reply:
Yes, please cherry-pick https://github.com/OCA/stock-logistics-workflow/pull/862 :pray:

Reply:
Thanks @simahawk , @grindtildeath for your comments. We are working on this.

Reply:
@simahawk, @grindtildeath, The PR has been updated.